cookbooks/aws-parallelcluster-platform/resources/gdrcopy/partial/_gdrcopy_common.rb

```diff
@@ -12,25 +12,37 @@
 # This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
 # See the License for the specific language governing permissions and limitations under the License.
 
-action :gdrcopy_installation do
-  gdrcopy_version = node['cluster']['nvidia']['gdrcopy']['version']
-  gdrcopy_tarball = "#{node['cluster']['sources_dir']}/gdrcopy-#{gdrcopy_version}.tar.gz"
-  gdrcopy_checksum = node['cluster']['nvidia']['gdrcopy']['sha256']
+property :gdrcopy_version, String, default: '2.3'
+property :gdrcopy_checksum, String, default: 'b85d15901889aa42de6c4a9233792af40dd94543e82abe0439e544c87fd79475'
+
+unified_mode true
+default_action :setup
+
+action :setup do
+  return unless gdrcopy_enabled?
+
+  # Save gdrcopy version for InSpec tests
+  node.default['cluster']['nvidia']['gdrcopy']['version'] = new_resource.gdrcopy_version
```

Comment:
When there are this kind of variables that are required by both code and inspec wouldn't be better to avoid defining them as properties of the resource?

The code would be cleaner and we would have the possibility to override them from chef node attributes at build image time.

Reply:
My reasoning is that this way we explicitly surface the dependencies.